solr/core/src/test/org/apache/solr/cloud/OverseerTest.java

```diff
@@ -758,7 +758,7 @@ private void waitForCollections(ZkStateReader stateReader, String... collections
     while (0 < maxIterations--) {
 
       final ClusterState state = stateReader.getClusterState();
-      Set<String> availableCollections = state.getCollectionsMap().keySet();
+      Set<String> availableCollections = Set.copyOf(state.getCollectionNames());
```

Comment:
`state.getCollectionNames()` might already return an unmodifiable view of keys from `immutableCollectionStates`. Is the `Set.copyOf()` necessary?

Reply:
state.getCollectionNames returns a Collection that just-so-happens to be a Set but may end up being a List in the future.  I didn't want to choose a Set on the first introduction of this method unless we think this choice is stable.

Since this spot here is merely a test, maybe I'll simply cast it to a Set.  If one day it becomes a List or something... whatever; easy enough to switch to Set.copyOf.  I wish the JDK had a convenient option to cast-or-create.  Didn't seem worth it in this class to hand-role a few lines of code for it.